{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Subject</th>\n",
       "      <th>Source1</th>\n",
       "      <th>Source2</th>\n",
       "      <th>Source3</th>\n",
       "      <th>Source4</th>\n",
       "      <th>Source5</th>\n",
       "      <th>Source6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Gene::</td>\n",
       "      <td>[IntAct] https://www.ebi.ac.uk/intact/</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Gene::1</td>\n",
       "      <td>[Bibliograpic sources] Collection of datasourc...</td>\n",
       "      <td>[Drugbank] https://www.drugbank.ca/atc</td>\n",
       "      <td>[STRING] https://string-db.org/</td>\n",
       "      <td>[Hetionet] Biomedical knowledge graph https://...</td>\n",
       "      <td>[GNBR] Data extracted from biomedical texts se...</td>\n",
       "      <td>[IntAct] https://www.ebi.ac.uk/intact/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Gene::1000</td>\n",
       "      <td>[Hetionet] Biomedical knowledge graph https://...</td>\n",
       "      <td>[STRING] https://string-db.org/</td>\n",
       "      <td>[GNBR] Data extracted from biomedical texts se...</td>\n",
       "      <td>[IntAct] https://www.ebi.ac.uk/intact/</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gene::10000</td>\n",
       "      <td>[STRING] https://string-db.org/</td>\n",
       "      <td>[DGIdb] http://www.dgidb.org/</td>\n",
       "      <td>[Hetionet] Biomedical knowledge graph https://...</td>\n",
       "      <td>[GNBR] Data extracted from biomedical texts se...</td>\n",
       "      <td>[IntAct] https://www.ebi.ac.uk/intact/</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gene::100000711</td>\n",
       "      <td>[GNBR] Data extracted from biomedical texts se...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Subject                                            Source1  \\\n",
       "0           Gene::             [IntAct] https://www.ebi.ac.uk/intact/   \n",
       "1          Gene::1  [Bibliograpic sources] Collection of datasourc...   \n",
       "2       Gene::1000  [Hetionet] Biomedical knowledge graph https://...   \n",
       "3      Gene::10000                    [STRING] https://string-db.org/   \n",
       "4  Gene::100000711  [GNBR] Data extracted from biomedical texts se...   \n",
       "\n",
       "                                  Source2  \\\n",
       "0                                     NaN   \n",
       "1  [Drugbank] https://www.drugbank.ca/atc   \n",
       "2         [STRING] https://string-db.org/   \n",
       "3           [DGIdb] http://www.dgidb.org/   \n",
       "4                                     NaN   \n",
       "\n",
       "                                             Source3  \\\n",
       "0                                                NaN   \n",
       "1                    [STRING] https://string-db.org/   \n",
       "2  [GNBR] Data extracted from biomedical texts se...   \n",
       "3  [Hetionet] Biomedical knowledge graph https://...   \n",
       "4                                                NaN   \n",
       "\n",
       "                                             Source4  \\\n",
       "0                                                NaN   \n",
       "1  [Hetionet] Biomedical knowledge graph https://...   \n",
       "2             [IntAct] https://www.ebi.ac.uk/intact/   \n",
       "3  [GNBR] Data extracted from biomedical texts se...   \n",
       "4                                                NaN   \n",
       "\n",
       "                                             Source5  \\\n",
       "0                                                NaN   \n",
       "1  [GNBR] Data extracted from biomedical texts se...   \n",
       "2                                                NaN   \n",
       "3             [IntAct] https://www.ebi.ac.uk/intact/   \n",
       "4                                                NaN   \n",
       "\n",
       "                                  Source6  \n",
       "0                                     NaN  \n",
       "1  [IntAct] https://www.ebi.ac.uk/intact/  \n",
       "2                                     NaN  \n",
       "3                                     NaN  \n",
       "4                                     NaN  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DRKG_PATH = \"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/embed/drkg.tsv\"\n",
    "drkg_data = pd.read_table(DRKG_PATH, sep = \"\\t\")\n",
    "#entity2src = pd.read_table(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/entity2src.tsv\",sep = \"\\t\",on_bad_lines='skip')\n",
    "entity2src = pd.read_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/entity2src.csv\",on_bad_lines='skip')\n",
    "entity2src.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def split_sources(df):\n",
    "    split_df = df.copy()\n",
    "    sources = [\"Source1\", \"Source2\", \"Source3\", \"Source4\", \"Source5\", \"Source6\"]\n",
    "    for source in sources: \n",
    "        split_df[source] = split_df[source].str.split()\n",
    "    return split_df\n",
    "        \n",
    "\n",
    "def get_source_name(split_df):\n",
    "    source_df = split_df.copy()\n",
    "    sources = [\"Source1\", \"Source2\", \"Source3\", \"Source4\", \"Source5\", \"Source6\"]\n",
    "    for source in sources : \n",
    "        source_df[source] = source_df[source].str[0]\n",
    "    return source_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Subject</th>\n",
       "      <th>Source1</th>\n",
       "      <th>Source2</th>\n",
       "      <th>Source3</th>\n",
       "      <th>Source4</th>\n",
       "      <th>Source5</th>\n",
       "      <th>Source6</th>\n",
       "      <th>SplitSubject</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Gene::</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Gene, ]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Gene::1</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>[Gene, 1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Gene::1000</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Gene, 1000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gene::10000</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[DGIdb]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Gene, 10000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gene::100000711</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Gene, 100000711]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Subject        Source1     Source2     Source3     Source4  \\\n",
       "0           Gene::       [IntAct]         NaN         NaN         NaN   \n",
       "1          Gene::1  [Bibliograpic  [Drugbank]    [STRING]  [Hetionet]   \n",
       "2       Gene::1000     [Hetionet]    [STRING]      [GNBR]    [IntAct]   \n",
       "3      Gene::10000       [STRING]     [DGIdb]  [Hetionet]      [GNBR]   \n",
       "4  Gene::100000711         [GNBR]         NaN         NaN         NaN   \n",
       "\n",
       "    Source5   Source6       SplitSubject  \n",
       "0       NaN       NaN           [Gene, ]  \n",
       "1    [GNBR]  [IntAct]          [Gene, 1]  \n",
       "2       NaN       NaN       [Gene, 1000]  \n",
       "3  [IntAct]       NaN      [Gene, 10000]  \n",
       "4       NaN       NaN  [Gene, 100000711]  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "split_df = split_sources(entity2src)\n",
    "source_df = get_source_name(split_df)\n",
    "\n",
    "def split_entity_types(df) : \n",
    "    df[\"SplitSubject\"] = df[\"Subject\"].str.split(\"::\")\n",
    "    \n",
    "def split_object_types(df): \n",
    "     df[\"SplitObject\"] = df[\"Object\"].str.split(\"::\")\n",
    "\n",
    "split_entity_types(source_df)\n",
    "source_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Subject</th>\n",
       "      <th>Source1</th>\n",
       "      <th>Source2</th>\n",
       "      <th>Source3</th>\n",
       "      <th>Source4</th>\n",
       "      <th>Source5</th>\n",
       "      <th>Source6</th>\n",
       "      <th>SplitSubject</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Gene::</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Gene, ]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Gene::1</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>[Gene, 1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Gene::1000</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Gene, 1000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gene::10000</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[DGIdb]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Gene, 10000]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gene::100000711</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Gene, 100000711]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Subject        Source1     Source2     Source3     Source4  \\\n",
       "0           Gene::       [IntAct]         NaN         NaN         NaN   \n",
       "1          Gene::1  [Bibliograpic  [Drugbank]    [STRING]  [Hetionet]   \n",
       "2       Gene::1000     [Hetionet]    [STRING]      [GNBR]    [IntAct]   \n",
       "3      Gene::10000       [STRING]     [DGIdb]  [Hetionet]      [GNBR]   \n",
       "4  Gene::100000711         [GNBR]         NaN         NaN         NaN   \n",
       "\n",
       "    Source5   Source6       SplitSubject  \n",
       "0       NaN       NaN           [Gene, ]  \n",
       "1    [GNBR]  [IntAct]          [Gene, 1]  \n",
       "2       NaN       NaN       [Gene, 1000]  \n",
       "3  [IntAct]       NaN      [Gene, 10000]  \n",
       "4       NaN       NaN  [Gene, 100000711]  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "split_entity_types(source_df)\n",
    "source_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Subject</th>\n",
       "      <th>Predicate</th>\n",
       "      <th>Object</th>\n",
       "      <th>Source1</th>\n",
       "      <th>Source2</th>\n",
       "      <th>Source3</th>\n",
       "      <th>Source4</th>\n",
       "      <th>Source5</th>\n",
       "      <th>Source6</th>\n",
       "      <th>SubjectType</th>\n",
       "      <th>SubjectID</th>\n",
       "      <th>ObjectType</th>\n",
       "      <th>ObjectID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Gene::6187</td>\n",
       "      <td>bioarx::HumGenHumGen:Gene:Gene</td>\n",
       "      <td>Gene::5566</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>Gene</td>\n",
       "      <td>6187</td>\n",
       "      <td>Gene</td>\n",
       "      <td>5566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Gene::6187</td>\n",
       "      <td>bioarx::HumGenHumGen:Gene:Gene</td>\n",
       "      <td>Gene::7412</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>Gene</td>\n",
       "      <td>6187</td>\n",
       "      <td>Gene</td>\n",
       "      <td>7412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Gene::6187</td>\n",
       "      <td>bioarx::HumGenHumGen:Gene:Gene</td>\n",
       "      <td>Gene::3845</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>Gene</td>\n",
       "      <td>6187</td>\n",
       "      <td>Gene</td>\n",
       "      <td>3845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gene::6187</td>\n",
       "      <td>bioarx::HumGenHumGen:Gene:Gene</td>\n",
       "      <td>Gene::3008</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>Gene</td>\n",
       "      <td>6187</td>\n",
       "      <td>Gene</td>\n",
       "      <td>3008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gene::6187</td>\n",
       "      <td>bioarx::HumGenHumGen:Gene:Gene</td>\n",
       "      <td>Gene::7083</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>Gene</td>\n",
       "      <td>6187</td>\n",
       "      <td>Gene</td>\n",
       "      <td>7083</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Subject                       Predicate      Object        Source1  \\\n",
       "0  Gene::6187  bioarx::HumGenHumGen:Gene:Gene  Gene::5566  [Bibliograpic   \n",
       "1  Gene::6187  bioarx::HumGenHumGen:Gene:Gene  Gene::7412  [Bibliograpic   \n",
       "2  Gene::6187  bioarx::HumGenHumGen:Gene:Gene  Gene::3845  [Bibliograpic   \n",
       "3  Gene::6187  bioarx::HumGenHumGen:Gene:Gene  Gene::3008  [Bibliograpic   \n",
       "4  Gene::6187  bioarx::HumGenHumGen:Gene:Gene  Gene::7083  [Bibliograpic   \n",
       "\n",
       "      Source2   Source3     Source4 Source5   Source6 SubjectType SubjectID  \\\n",
       "0  [Drugbank]  [STRING]  [Hetionet]  [GNBR]  [IntAct]        Gene      6187   \n",
       "1  [Drugbank]  [STRING]  [Hetionet]  [GNBR]  [IntAct]        Gene      6187   \n",
       "2  [Drugbank]  [STRING]  [Hetionet]  [GNBR]  [IntAct]        Gene      6187   \n",
       "3  [Drugbank]  [STRING]  [Hetionet]  [GNBR]  [IntAct]        Gene      6187   \n",
       "4  [Drugbank]  [STRING]  [Hetionet]  [GNBR]  [IntAct]        Gene      6187   \n",
       "\n",
       "  ObjectType ObjectID  \n",
       "0       Gene     5566  \n",
       "1       Gene     7412  \n",
       "2       Gene     3845  \n",
       "3       Gene     3008  \n",
       "4       Gene     7083  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def create_entity_type_df(df): \n",
    "    entity_type_df = df.copy()\n",
    "    entity_type_df[\"SubjectType\"] = entity_type_df[\"SplitSubject\"].str[0]\n",
    "    entity_type_df[\"SubjectID\"] = entity_type_df[\"SplitSubject\"].str[1]\n",
    "    return entity_type_df\n",
    "\n",
    "def create_object_type(df) : \n",
    "    entity_type_df = df.copy()\n",
    "    entity_type_df[\"ObjectType\"] = entity_type_df[\"SplitObject\"].str[0]\n",
    "    entity_type_df[\"ObjectID\"] = entity_type_df[\"SplitObject\"].str[1]\n",
    "    return entity_type_df\n",
    "\n",
    "entity_type_df =create_entity_type_df(source_df)\n",
    "entity_type_df.head()\n",
    "\n",
    "drkg = pd.merge(drkg_data, entity_type_df, on=\"Subject\")\n",
    "split_object_types(drkg)\n",
    "drkg = create_object_type(drkg)\n",
    "drkg = drkg.drop([\"SplitSubject\", \"SplitObject\"],axis = 1)\n",
    "drkg.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Subject</th>\n",
       "      <th>Predicate</th>\n",
       "      <th>Object</th>\n",
       "      <th>Source1</th>\n",
       "      <th>Source2</th>\n",
       "      <th>Source3</th>\n",
       "      <th>Source4</th>\n",
       "      <th>Source5</th>\n",
       "      <th>Source6</th>\n",
       "      <th>SubjectType</th>\n",
       "      <th>SubjectID</th>\n",
       "      <th>ObjectType</th>\n",
       "      <th>ObjectID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Gene::6187</td>\n",
       "      <td>bioarx::HumGenHumGen:Gene:Gene</td>\n",
       "      <td>Gene::5566</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>Gene</td>\n",
       "      <td>6187</td>\n",
       "      <td>Gene</td>\n",
       "      <td>5566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Gene::6187</td>\n",
       "      <td>bioarx::HumGenHumGen:Gene:Gene</td>\n",
       "      <td>Gene::7412</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>Gene</td>\n",
       "      <td>6187</td>\n",
       "      <td>Gene</td>\n",
       "      <td>7412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Gene::6187</td>\n",
       "      <td>bioarx::HumGenHumGen:Gene:Gene</td>\n",
       "      <td>Gene::3845</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>Gene</td>\n",
       "      <td>6187</td>\n",
       "      <td>Gene</td>\n",
       "      <td>3845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gene::6187</td>\n",
       "      <td>bioarx::HumGenHumGen:Gene:Gene</td>\n",
       "      <td>Gene::3008</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>Gene</td>\n",
       "      <td>6187</td>\n",
       "      <td>Gene</td>\n",
       "      <td>3008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gene::6187</td>\n",
       "      <td>bioarx::HumGenHumGen:Gene:Gene</td>\n",
       "      <td>Gene::7083</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[STRING]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>[IntAct]</td>\n",
       "      <td>Gene</td>\n",
       "      <td>6187</td>\n",
       "      <td>Gene</td>\n",
       "      <td>7083</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Subject                       Predicate      Object        Source1  \\\n",
       "0  Gene::6187  bioarx::HumGenHumGen:Gene:Gene  Gene::5566  [Bibliograpic   \n",
       "1  Gene::6187  bioarx::HumGenHumGen:Gene:Gene  Gene::7412  [Bibliograpic   \n",
       "2  Gene::6187  bioarx::HumGenHumGen:Gene:Gene  Gene::3845  [Bibliograpic   \n",
       "3  Gene::6187  bioarx::HumGenHumGen:Gene:Gene  Gene::3008  [Bibliograpic   \n",
       "4  Gene::6187  bioarx::HumGenHumGen:Gene:Gene  Gene::7083  [Bibliograpic   \n",
       "\n",
       "      Source2   Source3     Source4 Source5   Source6 SubjectType SubjectID  \\\n",
       "0  [Drugbank]  [STRING]  [Hetionet]  [GNBR]  [IntAct]        Gene      6187   \n",
       "1  [Drugbank]  [STRING]  [Hetionet]  [GNBR]  [IntAct]        Gene      6187   \n",
       "2  [Drugbank]  [STRING]  [Hetionet]  [GNBR]  [IntAct]        Gene      6187   \n",
       "3  [Drugbank]  [STRING]  [Hetionet]  [GNBR]  [IntAct]        Gene      6187   \n",
       "4  [Drugbank]  [STRING]  [Hetionet]  [GNBR]  [IntAct]        Gene      6187   \n",
       "\n",
       "  ObjectType ObjectID  \n",
       "0       Gene     5566  \n",
       "1       Gene     7412  \n",
       "2       Gene     3845  \n",
       "3       Gene     3008  \n",
       "4       Gene     7083  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "entity_type_df =create_entity_type_df(source_df)\n",
    "drkg = pd.merge(drkg_data, entity_type_df, on=\"Subject\")\n",
    "split_object_types(drkg)\n",
    "drkg = create_object_type(drkg)\n",
    "drkg = drkg.drop([\"SplitSubject\", \"SplitObject\"],axis = 1)\n",
    "drkg.head()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Split DRKG into different sources to handle entities more easily\n",
    "### Drugs are mostly contained in GNBR,Hetionet, Drugbank and DGIdb sources. \n",
    "### Diseases are mostly contained in GNBR,Hetionet, Drugbank sources. \n",
    "### Gene & Gene interaction info is stores in STRING, IntAct and Hetionet sources. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "Hetionet = drkg.loc[(drkg[\"Source1\"] == \"[Hetionet]\") | (drkg[\"Source2\"] == \"[Hetionet]\") | (drkg[\"Source3\"] == \"[Hetionet]\")\n",
    "            | (drkg[\"Source4\"] == \"[Hetionet]\")| (drkg[\"Source5\"] == \"[Hetionet]\") | (drkg[\"Source6\"] == \"[Hetionet]\")]\n",
    "\n",
    "DrugBank = drkg.loc[(drkg[\"Source1\"] == \"[Drugbank]\") | (drkg[\"Source2\"] == \"[Drugbank]\") | (drkg[\"Source3\"] == \"[Drugbank]\")\n",
    "            | (drkg[\"Source4\"] == \"[Drugbank]\")| (drkg[\"Source5\"] == \"[Drugbank]\") | (drkg[\"Source6\"] == \"[Drugbank]\")]\n",
    "\n",
    "GNBR = drkg.loc[(drkg[\"Source1\"] == \"[GNBR]\") | (drkg[\"Source2\"] == \"[GNBR]\") | (drkg[\"Source3\"] == \"[GNBR]\")\n",
    "            | (drkg[\"Source4\"] == \"[GNBR]\")| (drkg[\"Source5\"] == \"[GNBR]\") | (drkg[\"Source6\"] == \"[GNBR]\")]\n",
    "\n",
    "IntAct = drkg.loc[(drkg[\"Source1\"] == \"[IntAct]\") | (drkg[\"Source2\"] == \"[IntAct]\") | (drkg[\"Source3\"] == \"[IntAct]\")\n",
    "            | (drkg[\"Source4\"] == \"[IntAct]\")| (drkg[\"Source5\"] == \"[IntAct]\") | (drkg[\"Source6\"] == \"[IntAct]\")]\n",
    "\n",
    "STRING = drkg.loc[(drkg[\"Source1\"] == \"[STRING]\") | (drkg[\"Source2\"] == \"[STRING]\") | (drkg[\"Source3\"] == \"[STRING]\")\n",
    "            | (drkg[\"Source4\"] == \"[STRING]\")| (drkg[\"Source5\"] == \"[STRING]\") | (drkg[\"Source6\"] == \"[STRING]\")]\n",
    "\n",
    "DGIdb = drkg.loc[(drkg[\"Source1\"] == \"[DGIdb]\") | (drkg[\"Source2\"] == \"[DGIdb]\") | (drkg[\"Source3\"] == \"[DGIdb]\")\n",
    "            | (drkg[\"Source4\"] == \"[DGIdb]\")| (drkg[\"Source5\"] == \"[DGIdb]\") | (drkg[\"Source6\"] == \"[DGIdb]\")]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create a file containing drug-disease indications\n",
    "### Using the predicates indicating treatment from the relation glossary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "DrugBank_indication = DrugBank.loc[DrugBank[\"Predicate\"]== \"DRUGBANK::treats::Compound:Disease\"]\n",
    "Hetionet_indication = Hetionet.loc[Hetionet[\"Predicate\"] == \"Hetionet::CtD::Compound:Disease\"]\n",
    "GNBR_indication = GNBR.loc[(GNBR[\"Predicate\"] == \"GNBR::T::Compound:Disease\")| (GNBR[\"Predicate\"] == \"GNBR::Te::Compound:Disease\")]\n",
    "\n",
    "indications = pd.concat([DrugBank_indication,Hetionet_indication,GNBR_indication])\n",
    "indications = indications[[\"SubjectID\", \"ObjectID\"]]\n",
    "indications.to_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/output/drug-disease-indications.csv\",index= False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Side Effect Similarity from DRKG \n",
    "#### The strategy is to get the side effects using the relevant predicates showed in relation glossary and compute the Jaccard similarity scores. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "gnbr_drug_se = GNBR.loc[GNBR[\"Predicate\"] == \"GNBR::Sa::Compound:Disease\"]\n",
    "hetionet_drug_se = Hetionet.loc[Hetionet[\"Predicate\"] == \"Hetionet::CcSE::Compound:Side Effect\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "mesh_diseases= pd.read_table(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/MESH_diseases_vocab.tsv\", sep = \"\\t\")\n",
    "mesh_diseases[\"d\"] = mesh_diseases[\"d\"].str.upper()\n",
    "mesh_diseases[\"name\"] = mesh_diseases[\"name\"].str.lower()\n",
    "mesh_diseases = mesh_diseases.rename(columns={\"d\" : \"ObjectID\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "gnbr_drug_se = pd.merge(gnbr_drug_se,mesh_diseases,on = \"ObjectID\" )\n",
    "\n",
    "sider_drugs = pd.read_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/sider4_vocab.tsv\",sep = \"\\t\")\n",
    "sider_drugs = sider_drugs.rename(columns={\"umls_cui_from_meddra\" : \"ObjectID\",\"side_effect_name\": \"name\"})\n",
    "sider_drugs[\"name\"] = sider_drugs[\"name\"].str.lower()\n",
    "\n",
    "hetionet_drug_se = pd.merge(hetionet_drug_se,sider_drugs, on = \"ObjectID\")\n",
    "gnbr_drug_se = gnbr_drug_se.loc[gnbr_drug_se[\"Subject\"].str.contains(\"MESH\")==False]\n",
    "\n",
    "drug_se = pd.concat([gnbr_drug_se,hetionet_drug_se],ignore_index=True)\n",
    "drug_se = drug_se.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1985"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = drug_se[\"SubjectID\"]\n",
    "b = indications[\"SubjectID\"]\n",
    "intersect = set(a).intersection(b)\n",
    "len(intersect)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def se_similarity(df) : \n",
    "    import itertools\n",
    "    drug_se_dict = {k : g[\"ObjectID\"].tolist() for k,g in df.groupby(\"SubjectID\")}\n",
    "    sim_scores = []\n",
    "\n",
    "    for combination in itertools.combinations(intersect,2):\n",
    "        drug1 = combination[0]\n",
    "        drug2 = combination[1]\n",
    "\n",
    "        side_effects1 = drug_se_dict[drug1]\n",
    "        side_effects2 = drug_se_dict[drug2]\n",
    "\n",
    "        c = set(side_effects1).intersection(side_effects2)\n",
    "        u = set(side_effects1).union(side_effects2)\n",
    "\n",
    "        sim_score = len(c) / float(len(u))\n",
    "        sim_scores.append([drug1,drug2,sim_score])\n",
    "    drug_se_df = pd.DataFrame(sim_scores, columns = [\"Drug1\", \"Drug2\", \"SE-SIM\"])\n",
    "    return drug_se_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "se_sim = se_similarity(drug_se)\n",
    "se_sim.to_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/output/drugs-se-sim.csv\", index= False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fingeprint Similarity from DRKG\n",
    "### Drug SMILES representation is available in DRKG.tar file. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SubjectID</th>\n",
       "      <th>SMILES</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DB00006</td>\n",
       "      <td>CC[C@H](C)[C@H](NC(=O)[C@H](CCC(O)=O)NC(=O)[C@...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DB00007</td>\n",
       "      <td>CCNC(=O)[C@@H]1CCCN1C(=O)[C@H](CCCNC(N)=N)NC(=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DB00014</td>\n",
       "      <td>CC(C)C[C@H](NC(=O)[C@@H](COC(C)(C)C)NC(=O)[C@H...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DB00027</td>\n",
       "      <td>CC(C)C[C@@H](NC(=O)CNC(=O)[C@@H](NC=O)C(C)C)C(...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DB00035</td>\n",
       "      <td>NC(=O)CC[C@@H]1NC(=O)[C@H](CC2=CC=CC=C2)NC(=O)...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  SubjectID                                             SMILES\n",
       "0   DB00006  CC[C@H](C)[C@H](NC(=O)[C@H](CCC(O)=O)NC(=O)[C@...\n",
       "1   DB00007  CCNC(=O)[C@@H]1CCCN1C(=O)[C@H](CCCNC(N)=N)NC(=...\n",
       "2   DB00014  CC(C)C[C@H](NC(=O)[C@@H](COC(C)(C)C)NC(=O)[C@H...\n",
       "3   DB00027  CC(C)C[C@@H](NC(=O)CNC(=O)[C@@H](NC=O)C(C)C)C(...\n",
       "4   DB00035  NC(=O)CC[C@@H]1NC(=O)[C@H](CC2=CC=CC=C2)NC(=O)..."
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "smiles= pd.read_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/SMILES.csv\",on_bad_lines='skip')\n",
    "smiles[\"SMILES\"] = smiles[\"SMILES\"].str.replace(\"\\t\",\"\")\n",
    "smiles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Subject</th>\n",
       "      <th>Predicate</th>\n",
       "      <th>Object</th>\n",
       "      <th>Source1</th>\n",
       "      <th>Source2</th>\n",
       "      <th>Source3</th>\n",
       "      <th>Source4</th>\n",
       "      <th>Source5</th>\n",
       "      <th>Source6</th>\n",
       "      <th>SubjectType</th>\n",
       "      <th>SubjectID</th>\n",
       "      <th>ObjectType</th>\n",
       "      <th>ObjectID</th>\n",
       "      <th>name</th>\n",
       "      <th>SMILES</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Compound::DB00244</td>\n",
       "      <td>GNBR::Sa::Compound:Disease</td>\n",
       "      <td>Disease::MESH:D007674</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[DGIdb]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Compound</td>\n",
       "      <td>DB00244</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D007674</td>\n",
       "      <td>kidney diseases</td>\n",
       "      <td>NC1=CC(C(O)=O)=C(O)C=C1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Compound::DB00244</td>\n",
       "      <td>GNBR::Sa::Compound:Disease</td>\n",
       "      <td>Disease::MESH:D009395</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[DGIdb]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Compound</td>\n",
       "      <td>DB00244</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D009395</td>\n",
       "      <td>nephritis, interstitial</td>\n",
       "      <td>NC1=CC(C(O)=O)=C(O)C=C1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Compound::DB00244</td>\n",
       "      <td>GNBR::Sa::Compound:Disease</td>\n",
       "      <td>Disease::MESH:D010190</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[DGIdb]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Compound</td>\n",
       "      <td>DB00244</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D010190</td>\n",
       "      <td>pancreatic neoplasms</td>\n",
       "      <td>NC1=CC(C(O)=O)=C(O)C=C1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Compound::DB00244</td>\n",
       "      <td>GNBR::Sa::Compound:Disease</td>\n",
       "      <td>Disease::MESH:D019283</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[DGIdb]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Compound</td>\n",
       "      <td>DB00244</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D019283</td>\n",
       "      <td>pancreatitis, acute necrotizing</td>\n",
       "      <td>NC1=CC(C(O)=O)=C(O)C=C1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Compound::DB00244</td>\n",
       "      <td>GNBR::Sa::Compound:Disease</td>\n",
       "      <td>Disease::MESH:D001919</td>\n",
       "      <td>[Bibliograpic</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[DGIdb]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Compound</td>\n",
       "      <td>DB00244</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D001919</td>\n",
       "      <td>bradycardia</td>\n",
       "      <td>NC1=CC(C(O)=O)=C(O)C=C1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Subject                   Predicate                 Object  \\\n",
       "0  Compound::DB00244  GNBR::Sa::Compound:Disease  Disease::MESH:D007674   \n",
       "1  Compound::DB00244  GNBR::Sa::Compound:Disease  Disease::MESH:D009395   \n",
       "2  Compound::DB00244  GNBR::Sa::Compound:Disease  Disease::MESH:D010190   \n",
       "3  Compound::DB00244  GNBR::Sa::Compound:Disease  Disease::MESH:D019283   \n",
       "4  Compound::DB00244  GNBR::Sa::Compound:Disease  Disease::MESH:D001919   \n",
       "\n",
       "         Source1     Source2  Source3     Source4 Source5 Source6 SubjectType  \\\n",
       "0  [Bibliograpic  [Drugbank]  [DGIdb]  [Hetionet]  [GNBR]     NaN    Compound   \n",
       "1  [Bibliograpic  [Drugbank]  [DGIdb]  [Hetionet]  [GNBR]     NaN    Compound   \n",
       "2  [Bibliograpic  [Drugbank]  [DGIdb]  [Hetionet]  [GNBR]     NaN    Compound   \n",
       "3  [Bibliograpic  [Drugbank]  [DGIdb]  [Hetionet]  [GNBR]     NaN    Compound   \n",
       "4  [Bibliograpic  [Drugbank]  [DGIdb]  [Hetionet]  [GNBR]     NaN    Compound   \n",
       "\n",
       "  SubjectID ObjectType      ObjectID                             name  \\\n",
       "0   DB00244    Disease  MESH:D007674                  kidney diseases   \n",
       "1   DB00244    Disease  MESH:D009395          nephritis, interstitial   \n",
       "2   DB00244    Disease  MESH:D010190             pancreatic neoplasms   \n",
       "3   DB00244    Disease  MESH:D019283  pancreatitis, acute necrotizing   \n",
       "4   DB00244    Disease  MESH:D001919                      bradycardia   \n",
       "\n",
       "                    SMILES  \n",
       "0  NC1=CC(C(O)=O)=C(O)C=C1  \n",
       "1  NC1=CC(C(O)=O)=C(O)C=C1  \n",
       "2  NC1=CC(C(O)=O)=C(O)C=C1  \n",
       "3  NC1=CC(C(O)=O)=C(O)C=C1  \n",
       "4  NC1=CC(C(O)=O)=C(O)C=C1  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_smiles = pd.merge(drug_se, smiles, on = \"SubjectID\" )\n",
    "drug_smiles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1854"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = drug_smiles[\"SubjectID\"]\n",
    "b = drug_se[\"SubjectID\"]\n",
    "intersect = set(a).intersection(b)\n",
    "len(intersect)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "import oddt\n",
    "from oddt import toolkit\n",
    "from oddt import fingerprints\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import itertools\n",
    "\n",
    "input_dict = dict()\n",
    "for index,line in drug_smiles.iterrows():\n",
    "     id = line[\"SubjectID\"]\n",
    "\n",
    "     smiles = line[\"SMILES\"]\n",
    "     mol = oddt.toolkit.readstring(format=\"smi\",string=smiles)\n",
    "     fp = mol.calcfp(fptype=\"MACCS\").raw\n",
    "     input_dict[id] = fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "def tanimoto_score(fp1, fp2):\n",
    "    return np.sum(fp1 &  fp2) / np.sum(fp1 | fp2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "c = drug_smiles[\"SubjectID\"]\n",
    "intersect = set(intersect).intersection(c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "sim_values=[]\n",
    "for chemical1, chemical2 in itertools.combinations(input_dict.keys(),2):\n",
    "    TC= tanimoto_score(input_dict[chemical1], input_dict[chemical2])\n",
    "    if chemical1 != chemical2:\n",
    "        sim_values.append([chemical1, chemical2, TC])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "chem_sim_df = pd.DataFrame(sim_values, columns=['Drug1','Drug2','TC'])\n",
    "chem_sim_df.head()\n",
    "chem_sim_df.to_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/output/drugs-tc-sim.csv\", index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Target Sequence Similarity from DRKG\n",
    "### The strategy is to find the target genes using the relevant predicates from DRKG, then match them with their sequence representations taken from hugo database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ObjectID</th>\n",
       "      <th>seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6035</td>\n",
       "      <td>MALEKSLVRLLLLVLILLVLGWVQPSLGKESRAKKFQRQHMDSDSS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>635</td>\n",
       "      <td>MPPVGGKKAKKGILERLNAGEIVIGDGGFVFALEKRGYVKAGPWTP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23743</td>\n",
       "      <td>MAPAGRPGAKKGILERLESGEVVIGDGSFLITLEKRGYVKAGLWTP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6301</td>\n",
       "      <td>VLDLDLFRVDKGGDPALIRETQEKRFKDPGLVDQLVKADSEWRRCR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>645</td>\n",
       "      <td>AVKKIAIFGATGQTGLTTLAQAVQAGYEVTVLVRDSSRLPSEGPRP...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ObjectID                                                seq\n",
       "0     6035  MALEKSLVRLLLLVLILLVLGWVQPSLGKESRAKKFQRQHMDSDSS...\n",
       "1      635  MPPVGGKKAKKGILERLNAGEIVIGDGGFVFALEKRGYVKAGPWTP...\n",
       "2    23743  MAPAGRPGAKKGILERLESGEVVIGDGSFLITLEKRGYVKAGLWTP...\n",
       "3     6301  VLDLDLFRVDKGGDPALIRETQEKRFKDPGLVDQLVKADSEWRRCR...\n",
       "4      645  AVKKIAIFGATGQTGLTTLAQAVQAGYEVTVLVRDSSRLPSEGPRP..."
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def fasta2seq(lines):\n",
    "    lines = lines[lines.index('\\n')+1:]\n",
    "    lines =lines.replace('\\n','')\n",
    "    return lines\n",
    "\n",
    "target_sequence = pd.read_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/targetseq.csv\")\n",
    "target_sequence.seq =target_sequence.seq.map(fasta2seq)\n",
    "target_sequence = target_sequence.rename(columns={\"geneid\": \"ObjectID\"})\n",
    "target_sequence[\"ObjectID\"] = target_sequence[\"ObjectID\"].astype(\"string\")\n",
    "target_sequence.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "gnbr_drug_target = GNBR.loc[(GNBR[\"Predicate\"] == \"GNBR::B::Compound:Gene\") | \n",
    "                        (GNBR[\"Predicate\"] == \"GNBR::E+::Compound:Gene\") |\n",
    "                        (GNBR[\"Predicate\"] == \"GNBR::E-::Compound:Gene\") | \n",
    "                         (GNBR[\"Predicate\"] == \"GNBR::E::Compound:Gene\") |\n",
    "                          (GNBR[\"Predicate\"] == \"GNBR::N::Compound:Gene\") |\n",
    "                           (GNBR[\"Predicate\"] == \"GNBR::O::Compound:Gene\") ]\n",
    "\n",
    "hetionet_drug_target = Hetionet.loc[(Hetionet[\"Predicate\"] == \"Hetionet::CbG::Compound:Gene\" )|\n",
    "                                (Hetionet[\"Predicate\"] == \"Hetionet::CdG::Compound:Gene\")|\n",
    "                                (Hetionet[\"Predicate\"] == \"Hetionet::CuG::Compound:Gene\")]\n",
    "\n",
    "drugbank_drug_target = DrugBank.loc[DrugBank[\"Predicate\"] == \"DRUGBANK::target::Compound:Gene\"]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "drug_target_seq = pd.concat([hetionet_drug_target,gnbr_drug_target,drugbank_drug_target],ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "drug_target_sequence = pd.merge(target_sequence,drug_target_seq, on = \"ObjectID\")\n",
    "drug_target_sequence.head()\n",
    "d = drug_target_sequence[\"SubjectID\"]\n",
    "intersect = set(intersect).intersection(d)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "import itertools\n",
    "import numpy as np\n",
    "\n",
    "def matrix(a, b, match_score=3, gap_cost=2):\n",
    "    H = np.zeros((len(a) + 1, len(b) + 1), np.int32)\n",
    "\n",
    "    for i, j in itertools.product(range(1, H.shape[0]), range(1, H.shape[1])):\n",
    "        match = H[i - 1, j - 1] + (match_score if a[i - 1] == b[j - 1] else - match_score)\n",
    "        delete = H[i - 1, j] - gap_cost\n",
    "        insert = H[i, j - 1] - gap_cost\n",
    "        H[i, j] = max(match, delete, insert, 0)\n",
    "    return H"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "def target_seq_similarity(df):\n",
    "    target_sequence_list = {k: g[\"seq\"].tolist() for k,g in df.groupby(\"SubjectID\")} \n",
    "    scores = []\n",
    "\n",
    "    for combination in itertools.combinations(intersect,2): \n",
    "        drug1 = combination[0]\n",
    "        drug2 = combination[1]\n",
    "        if not (drug1 in target_sequence_list and drug2 in target_sequence_list) : continue\n",
    "        target_list1 = target_sequence_list[drug1]\n",
    "        target_list2 = target_sequence_list[drug2]\n",
    "        allscores = []\n",
    "        for target1 in sorted(target_list1): \n",
    "            genescores = []\n",
    "            for target2 in sorted(target_list2): \n",
    "                target1 = str(target1)\n",
    "                target2 = str(target2)\n",
    "                if target1 == target2:\n",
    "                    score = 1.0 \n",
    "                else: \n",
    "                    score = np.mean(matrix(target1,target2))\n",
    "        scores.append([drug1,drug2,score])\n",
    "    df = pd.DataFrame(scores, columns = [\"Drug1\", \"Drug2\", \"TARGETSEQ-SIM\"])\n",
    "    return df   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb Cell 33\u001b[0m in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> <a href='vscode-notebook-cell:/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb#X44sZmlsZQ%3D%3D?line=0'>1</a>\u001b[0m targetseq \u001b[39m=\u001b[39m target_seq_similarity(drug_target_sequence)\n",
      "\u001b[1;32m/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb Cell 33\u001b[0m in \u001b[0;36mtarget_seq_similarity\u001b[0;34m(df)\u001b[0m\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb#X44sZmlsZQ%3D%3D?line=17'>18</a>\u001b[0m                 score \u001b[39m=\u001b[39m \u001b[39m1.0\u001b[39m \n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb#X44sZmlsZQ%3D%3D?line=18'>19</a>\u001b[0m             \u001b[39melse\u001b[39;00m: \n\u001b[0;32m---> <a href='vscode-notebook-cell:/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb#X44sZmlsZQ%3D%3D?line=19'>20</a>\u001b[0m                 score \u001b[39m=\u001b[39m np\u001b[39m.\u001b[39mmean(matrix(target1,target2))\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb#X44sZmlsZQ%3D%3D?line=20'>21</a>\u001b[0m     scores\u001b[39m.\u001b[39mappend([drug1,drug2,score])\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb#X44sZmlsZQ%3D%3D?line=21'>22</a>\u001b[0m df \u001b[39m=\u001b[39m pd\u001b[39m.\u001b[39mDataFrame(scores, columns \u001b[39m=\u001b[39m [\u001b[39m\"\u001b[39m\u001b[39mDrug1\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39mDrug2\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39m\"\u001b[39m\u001b[39mTARGETSEQ-SIM\u001b[39m\u001b[39m\"\u001b[39m])\n",
      "\u001b[1;32m/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb Cell 33\u001b[0m in \u001b[0;36mmatrix\u001b[0;34m(a, b, match_score, gap_cost)\u001b[0m\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb#X44sZmlsZQ%3D%3D?line=6'>7</a>\u001b[0m \u001b[39mfor\u001b[39;00m i, j \u001b[39min\u001b[39;00m itertools\u001b[39m.\u001b[39mproduct(\u001b[39mrange\u001b[39m(\u001b[39m1\u001b[39m, H\u001b[39m.\u001b[39mshape[\u001b[39m0\u001b[39m]), \u001b[39mrange\u001b[39m(\u001b[39m1\u001b[39m, H\u001b[39m.\u001b[39mshape[\u001b[39m1\u001b[39m])):\n\u001b[1;32m      <a href='vscode-notebook-cell:/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb#X44sZmlsZQ%3D%3D?line=7'>8</a>\u001b[0m     match \u001b[39m=\u001b[39m H[i \u001b[39m-\u001b[39m \u001b[39m1\u001b[39m, j \u001b[39m-\u001b[39m \u001b[39m1\u001b[39m] \u001b[39m+\u001b[39m (match_score \u001b[39mif\u001b[39;00m a[i \u001b[39m-\u001b[39m \u001b[39m1\u001b[39m] \u001b[39m==\u001b[39m b[j \u001b[39m-\u001b[39m \u001b[39m1\u001b[39m] \u001b[39melse\u001b[39;00m \u001b[39m-\u001b[39m match_score)\n\u001b[0;32m----> <a href='vscode-notebook-cell:/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb#X44sZmlsZQ%3D%3D?line=8'>9</a>\u001b[0m     delete \u001b[39m=\u001b[39m H[i \u001b[39m-\u001b[39;49m \u001b[39m1\u001b[39;49m, j] \u001b[39m-\u001b[39;49m gap_cost\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb#X44sZmlsZQ%3D%3D?line=9'>10</a>\u001b[0m     insert \u001b[39m=\u001b[39m H[i, j \u001b[39m-\u001b[39m \u001b[39m1\u001b[39m] \u001b[39m-\u001b[39m gap_cost\n\u001b[1;32m     <a href='vscode-notebook-cell:/Users/elifozkan/Desktop/translator-openpredict/src/drkg_model/get_data.ipynb#X44sZmlsZQ%3D%3D?line=10'>11</a>\u001b[0m     H[i, j] \u001b[39m=\u001b[39m \u001b[39mmax\u001b[39m(match, delete, insert, \u001b[39m0\u001b[39m)\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "targetseq = target_seq_similarity(drug_target_sequence)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "targetseq.to_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/output/drugs-TARGETSEQ-sim.csv\",index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PPI-Based Similarity from DRKG\n",
    "### Using the target genes, we first find the protein interactions using the relevant predicates from drkg, then form a graph using the nodes obtained from DRKG. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "ppi_gnbr = GNBR.loc[(GNBR[\"Predicate\"] == \"GNBR::B::Gene:Gene\") | \n",
    "                        (GNBR[\"Predicate\"] == \"GNBR::E+::Geme:Gene\") |\n",
    "                         (GNBR[\"Predicate\"] == \"GNBR::E::Gene:Gene\") |\n",
    "                          (GNBR[\"Predicate\"] == \"GNBR::I::Gene:Gene\") |\n",
    "                           (GNBR[\"Predicate\"] == \"GNBR::Rg::Gene:Gene\") |\n",
    "                           (GNBR[\"Predicate\"] == \"GNBR::Q::Gene:Gene\")|\n",
    "                           (GNBR[\"Predicate\"] == \"GNBR::V+::Gene:Gene\")|\n",
    "                           (GNBR[\"Predicate\"] == \"GNBR::W::Gene:Gene\")]\n",
    "\n",
    "ppi_hetionet = Hetionet.loc[Hetionet[\"Predicate\"] == \"Hetionet::GIG::Gene:Gene\"]\n",
    "\n",
    "ppi_intact = IntAct.loc[(IntAct[\"Predicate\"] == \"INTACT::ASSOCIATION::Gene:Gene\" )|\n",
    "                        (IntAct[\"Predicate\"] == \"INTACT::DIRECT INTERACTION::Gene:Gene\" ) |\n",
    "                        (IntAct[\"Predicate\"] == \"INTACT::PHYSICAL ASSOCIATION::Gene:Gene\" )]\n",
    "\n",
    "ppi_string = STRING.loc[(STRING[\"Predicate\"] == \"STRING::ACTIVATION::Gene:Gene\")|\n",
    "                        (STRING[\"Predicate\"] == \"STRING::BINDING::Gene:Gene\") |\n",
    "                        (STRING[\"Predicate\"] == \"STRING::CATALYSIS::Gene:Gene\")|\n",
    "                        (STRING[\"Predicate\"] == \"STRING::INHIBITION::Gene:Gene\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>SubjectID</th>\n",
       "      <th>ObjectID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>6187</td>\n",
       "      <td>2248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>563</td>\n",
       "      <td>9370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>6584</td>\n",
       "      <td>5243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>10747</td>\n",
       "      <td>50639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>8573</td>\n",
       "      <td>10716</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246144</th>\n",
       "      <td>258610</td>\n",
       "      <td>9808</td>\n",
       "      <td>6184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246145</th>\n",
       "      <td>258611</td>\n",
       "      <td>9808</td>\n",
       "      <td>7157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246146</th>\n",
       "      <td>258612</td>\n",
       "      <td>9808</td>\n",
       "      <td>84675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246147</th>\n",
       "      <td>258613</td>\n",
       "      <td>9808</td>\n",
       "      <td>84676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246148</th>\n",
       "      <td>258614</td>\n",
       "      <td>9881</td>\n",
       "      <td>27185</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>246149 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         index SubjectID ObjectID\n",
       "0            0      6187     2248\n",
       "1            1       563     9370\n",
       "2            2      6584     5243\n",
       "3            3     10747    50639\n",
       "4            5      8573    10716\n",
       "...        ...       ...      ...\n",
       "246144  258610      9808     6184\n",
       "246145  258611      9808     7157\n",
       "246146  258612      9808    84675\n",
       "246147  258613      9808    84676\n",
       "246148  258614      9881    27185\n",
       "\n",
       "[246149 rows x 3 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ppi = pd.concat([ppi_gnbr,ppi_hetionet,ppi_intact],ignore_index=True)\n",
    "ppi = ppi[[\"SubjectID\",\"ObjectID\"]]\n",
    "ppi = ppi.drop_duplicates().reset_index()\n",
    "ppi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "import networkx as nx \n",
    "ppi_net = pd.read_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/ppi_network.csv\")\n",
    "G = nx.Graph()\n",
    "for i in range(0,ppi_net.shape[0]): \n",
    "    G.add_edge(ppi_net[\"geneid1\"][i],ppi_net[\"geneid2\"][i])\n",
    "ppi_n=nx.shortest_path_length(G)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "def grapDistance(ppi, target1, target2):\n",
    "    \"\"\"\n",
    "    return the shortest path between two proteins in the PPI network\n",
    "    ppi : dictonary that contains distance of PPI \n",
    "    target1 : first protein name\n",
    "    target2 : second protein name\n",
    "    \"\"\"\n",
    "    maxValue = 9999\n",
    "    if target1 not in ppi:\n",
    "        return maxValue\n",
    "    else:\n",
    "        if target2 not in ppi[target1]:\n",
    "            return maxValue\n",
    "        else:\n",
    "            return ppi[target1][target2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "import math\n",
    "target_gene_list = {k: g[\"ObjectID\"].tolist() for k,g in drug_target_seq.groupby(\"SubjectID\")}\n",
    "values = []\n",
    "\n",
    "A = 0.9\n",
    "b = 1\n",
    "for comb in itertools.combinations(intersect,2) :\n",
    "    drug1 = comb[0]\n",
    "    drug2 = comb[1]\n",
    "    if not(drug1 in target_gene_list and drug2 in target_gene_list) : continue\n",
    "    targetList1 = target_gene_list[drug1]\n",
    "    targetList2 = target_gene_list[drug2]\n",
    "    allscores =[]\n",
    "    for target1 in sorted(targetList1):\n",
    "        genescores = []\n",
    "        for target2 in sorted(targetList2):\n",
    "            target1 =str(target1)\n",
    "            target2 =str(target2)    \n",
    "            if target1 == target2:\n",
    "                score=1.0\n",
    "            else:\n",
    "                score = A*math.exp(-b* grapDistance(ppi_net, target1, target2))\n",
    "            genescores.append(score)\n",
    "    # add maximal values between the two lists of associated genes \n",
    "    allscores.append(max(genescores))\n",
    "    if len(allscores) ==0: continue\n",
    "    #average the maximal scores \n",
    "    maxScore =np.mean(allscores)\n",
    "    if maxScore >= 0:\n",
    "        values.append([drug1, drug2, maxScore])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "drug_ppi_df = pd.DataFrame(values, columns =['Drug1','Drug2','PPI-SIM'])\n",
    "drug_ppi_df.to_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/output/drug-ppi-sim.csv\", index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GO-Based Similarity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>drugid</th>\n",
       "      <th>g</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DB00122</td>\n",
       "      <td>GO_0043083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DB00122</td>\n",
       "      <td>GO_0005576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DB00122</td>\n",
       "      <td>GO_0005615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DB00122</td>\n",
       "      <td>GO_0005634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DB00122</td>\n",
       "      <td>GO_0005794</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    drugid           g\n",
       "0  DB00122  GO_0043083\n",
       "1  DB00122  GO_0005576\n",
       "2  DB00122  GO_0005615\n",
       "3  DB00122  GO_0005634\n",
       "4  DB00122  GO_0005794"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_go = pd.read_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/go-sim.csv\")\n",
    "drug_go[\"g\"] = drug_go[\"g\"].str.replace(\"go:\", \"GO_\")\n",
    "drug_go.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Drug1</th>\n",
       "      <th>Drug2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://purl.obolibrary.org/obo/DB00122</td>\n",
       "      <td>http://purl.obolibrary.org/obo/DB00677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>http://purl.obolibrary.org/obo/DB00122</td>\n",
       "      <td>http://purl.obolibrary.org/obo/DB00119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>http://purl.obolibrary.org/obo/DB00122</td>\n",
       "      <td>http://purl.obolibrary.org/obo/DB00132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>http://purl.obolibrary.org/obo/DB00122</td>\n",
       "      <td>http://purl.obolibrary.org/obo/DB00114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>http://purl.obolibrary.org/obo/DB00122</td>\n",
       "      <td>http://purl.obolibrary.org/obo/DB00116</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Drug1  \\\n",
       "0  http://purl.obolibrary.org/obo/DB00122   \n",
       "1  http://purl.obolibrary.org/obo/DB00122   \n",
       "2  http://purl.obolibrary.org/obo/DB00122   \n",
       "3  http://purl.obolibrary.org/obo/DB00122   \n",
       "4  http://purl.obolibrary.org/obo/DB00122   \n",
       "\n",
       "                                    Drug2  \n",
       "0  http://purl.obolibrary.org/obo/DB00677  \n",
       "1  http://purl.obolibrary.org/obo/DB00119  \n",
       "2  http://purl.obolibrary.org/obo/DB00132  \n",
       "3  http://purl.obolibrary.org/obo/DB00114  \n",
       "4  http://purl.obolibrary.org/obo/DB00116  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "target_gene_list = {k: g[\"g\"].tolist() for k,g in drug_go.groupby(\"drugid\")}\n",
    "rows = []\n",
    "for comb in itertools.combinations(drug_go[\"drugid\"].unique(),2):\n",
    "    drug1 = comb[0]\n",
    "    drug2 = comb[1]\n",
    "    rows.append(['http://purl.obolibrary.org/obo/' + str(drug1),'http://purl.obolibrary.org/obo/'+str(drug2)])\n",
    "\n",
    "\n",
    "drug_query_df = pd.DataFrame(rows, columns =['Drug1','Drug2'])\n",
    "drug_query_df.to_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/drug.gene.go.query\",sep='\\t', header=False, index=False)\n",
    "drug_query_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "! if [ ! -f '/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/go.owl' ]; then gunzip '/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/go.owl'; fi "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------------------------------------\n",
      "\tSemantic Measures Library Toolkit 0.9.4c\n",
      "----------------------------------------------------------------------\n",
      "Toolkit dedicated to Semantic Measures computation and studies.\n",
      "Website: http://www.semantic-measures-library.org\n",
      "Developer: Sébastien Harispe\n",
      "\n",
      "Please cite: \n",
      "The Semantic Measures Library and Toolkit: \n",
      "fast computation of semantic similarity and relatedness \n",
      "using biomedical ontologies.\n",
      "Sébastien Harispe*, Sylvie Ranwez, Stefan Janaqi, Jacky Montmain\n",
      "Bioinformatics 2013; doi: 10.1093/bioinformatics/btt581\n",
      "----------------------------------------------------------------------\n",
      "Args [-t, sm]\n",
      "Args Tool    [-xmlconf, /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/sml.gene.go.conf]\n",
      "Loading SM Tool\n",
      "----------------------------------------------------------------------\n",
      "\tSemantic Measures Computer 0.9.4c\n",
      "----------------------------------------------------------------------\n",
      "Tool used to compute Semantic similarity/relatedness considering an\n",
      "XML configuration file or a specific profile dedicated to an ontology.\n",
      "\n",
      "---------------------------------------------------------------\n",
      " Processing XML configuration /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/sml.gene.go.conf\n",
      "---------------------------------------------------------------\n",
      "Loading XML conf from : /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/sml.gene.go.conf\n",
      "Setting maximal number of threads to 4\n",
      "add namespace prefix : gene ref : http://purl.obolibrary.org/obo/gene_\n",
      "add namespace prefix : HP ref : http://purl.obolibrary.org/obo/HP_\n",
      "add namespace prefix : MP ref : http://purl.obolibrary.org/obo/MP_\n",
      "add namespace prefix : GO ref : http://purl.obolibrary.org/obo/GO_\n",
      "add namespace prefix : OMIM ref : http://bio2rdf.org/omim:\n",
      "add namespace prefix : DRUGBANK ref : http://bio2rdf.org/drugbank:\n",
      "generic configuration loaded \n",
      "Loading sm Configuration : /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/sml.gene.go.conf\n",
      "1 IC configurations loaded \n",
      "1 pairwise measure configurations loaded \n",
      "1 groupwise measure configurations  loaded \n",
      "Configuration loaded... \n",
      "Loading Graph http://purl.obolibrary.org/obo/gene_\n",
      "-------------------------------------\n",
      " Loading DATA\n",
      "-------------------------------------\n",
      "-------------------------------------\n",
      " RDF Loader\n",
      "-------------------------------------\n",
      "Populate graph http://purl.obolibrary.org/obo/gene_ from /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/go.owl\n",
      "Parser loaded for: RDF/XML (mimeTypes=application/rdf+xml, application/xml; ext=rdf, rdfs, owl, xml)\n",
      "Parsing RDF file...\n",
      "Start Process\n",
      "100000 statements already loaded\n",
      "Number of vertices: 41039\n",
      "Number of edges   : 100000\n",
      "Ending Process 130955 statements loaded \n",
      "vertices: 51180\n",
      "edges   : 130955\n",
      "Skipped (statement involving non URI ressources) : 1291679\n",
      "Graph http://purl.obolibrary.org/obo/gene_ populated by RDF data \n",
      "-------------------------------------\n",
      "-------------------------------------\n",
      "Loading Annotations using TSV loader.\n",
      "-------------------------------------\n",
      "file            /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/go-sim.csv\n",
      "Skipping header false\n",
      "predicate       'http://www.w3.org/1999/02/22-rdf-syntax-ns#type'\n",
      "Number of lines skipped   250001\n",
      "Number of lines processed 0\n",
      "Number of statements loaded 0\n",
      "TSV specification loaded.\n",
      "-------------------------------------\n",
      "-------------------------------------\n",
      " Applying actions\n",
      "-------------------------------------\n",
      "-------------------------------------\n",
      "Rerooting\n",
      "-------------------------------------\n",
      "Fetching root node, uri: __FICTIVE__\n",
      "No root node explicitly specified using 'root_uri' parameter. Set root : http://www.w3.org/2002/07/owl#Thing\n",
      "Rooting taxonomic Graph using http://www.w3.org/2002/07/owl#Thing\n",
      "Number of roots 7823\n",
      "Starting process from 34965 vertices\n",
      "isDag : true\n",
      "Number of roots detected: 7823\n",
      "Rooting performed using http://www.w3.org/2002/07/owl#Thing as root 7823 edges created\n",
      "Number of roots 1\n",
      "isDag : true\n",
      "Rerooting performed\n",
      "-------------------------------------\n",
      "Actions performed\n",
      "-------------------------------------\n",
      "Retrieving the graph http://purl.obolibrary.org/obo/gene_\n",
      "Graph information:\n",
      "http://purl.obolibrary.org/obo/gene_\n",
      "Vertices\n",
      "\tTotal   : 51181  {e.g. http://purl.obolibrary.org/obo/GO_0099593}\n",
      "Edges \t  : 138778\n",
      "\n",
      "\n",
      "Computing classes...\n",
      "================================================================\n",
      "Loading Semantic Measures Engine for graph http://purl.obolibrary.org/obo/gene_\n",
      "================================================================\n",
      "Graph Info: \n",
      "http://purl.obolibrary.org/obo/gene_\n",
      "Vertices\n",
      "\tTotal   : 51181  {e.g. http://purl.obolibrary.org/obo/GO_0099593}\n",
      "Edges \t  : 138778\n",
      "\n",
      "\n",
      "---------------------------------------------------------------\n",
      "Pre-processing\n",
      "---------------------------------------------------------------\n",
      "Classes  : 51069\n",
      "Instances Accessor loaded: false\n",
      "---------------------------------------------------------------\n",
      "Inferences \n",
      "---------------------------------------------------------------\n",
      "Inferring ancestors...\n",
      "Inferring descendants...\n"
     ]
    }
   ],
   "source": [
    "!java -jar /Users/elifozkan/Desktop/sml-toolkit-0.9.4c.jar -t sm -xmlconf /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/sml.gene.go.conf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/v7/vrxjycqx6736th34cms70szm0000gn/T/ipykernel_2357/1305721123.py:3: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  go_sim_df.Drug1 = go_sim_df.Drug1.str.replace('http://purl.obolibrary.org/obo/','')\n",
      "/var/folders/v7/vrxjycqx6736th34cms70szm0000gn/T/ipykernel_2357/1305721123.py:4: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  go_sim_df.Drug2 = go_sim_df.Drug2.str.replace('http://purl.obolibrary.org/obo/','')\n"
     ]
    }
   ],
   "source": [
    "go_sim_df = pd.read_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/drug.gene.go.sim.out\",sep='\\t')\n",
    "go_sim_df.rename(columns={'e1':'Drug1','e2':'Drug2','bma':'GO-SIM'}, inplace=True)\n",
    "go_sim_df.Drug1 = go_sim_df.Drug1.str.replace('http://purl.obolibrary.org/obo/','')\n",
    "go_sim_df.Drug2 = go_sim_df.Drug2.str.replace('http://purl.obolibrary.org/obo/','')\n",
    "go_sim_df.to_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/output/drug-go-sim.csv\",index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MESH Similarity for Diseases from DRKG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "GNBR_diseases = GNBR.loc[GNBR[\"SubjectType\"] == \"Disease\"]\n",
    "Hetionet_diseases = Hetionet.loc[Hetionet[\"SubjectType\"] == \"Disease\"]\n",
    "DrugBank_diseases = DrugBank.loc[DrugBank[\"SubjectType\"] == \"Disease\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Subject</th>\n",
       "      <th>Predicate</th>\n",
       "      <th>Object</th>\n",
       "      <th>Source1</th>\n",
       "      <th>Source2</th>\n",
       "      <th>Source3</th>\n",
       "      <th>Source4</th>\n",
       "      <th>Source5</th>\n",
       "      <th>Source6</th>\n",
       "      <th>SubjectType</th>\n",
       "      <th>SubjectID</th>\n",
       "      <th>ObjectType</th>\n",
       "      <th>ObjectID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4510005</th>\n",
       "      <td>Disease::MESH:D000506</td>\n",
       "      <td>Hetionet::DdG::Disease:Gene</td>\n",
       "      <td>Gene::26059</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D000506</td>\n",
       "      <td>Gene</td>\n",
       "      <td>26059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4510006</th>\n",
       "      <td>Disease::MESH:D000506</td>\n",
       "      <td>Hetionet::DdG::Disease:Gene</td>\n",
       "      <td>Gene::84445</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D000506</td>\n",
       "      <td>Gene</td>\n",
       "      <td>84445</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4510007</th>\n",
       "      <td>Disease::MESH:D000506</td>\n",
       "      <td>Hetionet::DdG::Disease:Gene</td>\n",
       "      <td>Gene::9632</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D000506</td>\n",
       "      <td>Gene</td>\n",
       "      <td>9632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4510008</th>\n",
       "      <td>Disease::MESH:D000506</td>\n",
       "      <td>Hetionet::DdG::Disease:Gene</td>\n",
       "      <td>Gene::8774</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D000506</td>\n",
       "      <td>Gene</td>\n",
       "      <td>8774</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4510009</th>\n",
       "      <td>Disease::MESH:D000506</td>\n",
       "      <td>Hetionet::DdG::Disease:Gene</td>\n",
       "      <td>Gene::131578</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D000506</td>\n",
       "      <td>Gene</td>\n",
       "      <td>131578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4545465</th>\n",
       "      <td>Disease::MESH:D005923</td>\n",
       "      <td>Hetionet::DaG::Disease:Gene</td>\n",
       "      <td>Gene::81</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D005923</td>\n",
       "      <td>Gene</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4545466</th>\n",
       "      <td>Disease::MESH:D005923</td>\n",
       "      <td>Hetionet::DaG::Disease:Gene</td>\n",
       "      <td>Gene::7436</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D005923</td>\n",
       "      <td>Gene</td>\n",
       "      <td>7436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4545467</th>\n",
       "      <td>Disease::MESH:D005923</td>\n",
       "      <td>Hetionet::DaG::Disease:Gene</td>\n",
       "      <td>Gene::1286</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D005923</td>\n",
       "      <td>Gene</td>\n",
       "      <td>1286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4545468</th>\n",
       "      <td>Disease::MESH:D005923</td>\n",
       "      <td>Hetionet::DaG::Disease:Gene</td>\n",
       "      <td>Gene::5054</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D005923</td>\n",
       "      <td>Gene</td>\n",
       "      <td>5054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4545469</th>\n",
       "      <td>Disease::MESH:D005923</td>\n",
       "      <td>Hetionet::DaG::Disease:Gene</td>\n",
       "      <td>Gene::4868</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D005923</td>\n",
       "      <td>Gene</td>\n",
       "      <td>4868</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>90128 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Subject                    Predicate        Object  \\\n",
       "4510005  Disease::MESH:D000506  Hetionet::DdG::Disease:Gene   Gene::26059   \n",
       "4510006  Disease::MESH:D000506  Hetionet::DdG::Disease:Gene   Gene::84445   \n",
       "4510007  Disease::MESH:D000506  Hetionet::DdG::Disease:Gene    Gene::9632   \n",
       "4510008  Disease::MESH:D000506  Hetionet::DdG::Disease:Gene    Gene::8774   \n",
       "4510009  Disease::MESH:D000506  Hetionet::DdG::Disease:Gene  Gene::131578   \n",
       "...                        ...                          ...           ...   \n",
       "4545465  Disease::MESH:D005923  Hetionet::DaG::Disease:Gene      Gene::81   \n",
       "4545466  Disease::MESH:D005923  Hetionet::DaG::Disease:Gene    Gene::7436   \n",
       "4545467  Disease::MESH:D005923  Hetionet::DaG::Disease:Gene    Gene::1286   \n",
       "4545468  Disease::MESH:D005923  Hetionet::DaG::Disease:Gene    Gene::5054   \n",
       "4545469  Disease::MESH:D005923  Hetionet::DaG::Disease:Gene    Gene::4868   \n",
       "\n",
       "            Source1     Source2 Source3 Source4 Source5 Source6 SubjectType  \\\n",
       "4510005  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4510006  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4510007  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4510008  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4510009  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "...             ...         ...     ...     ...     ...     ...         ...   \n",
       "4545465  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4545466  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4545467  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4545468  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4545469  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "\n",
       "            SubjectID ObjectType ObjectID  \n",
       "4510005  MESH:D000506       Gene    26059  \n",
       "4510006  MESH:D000506       Gene    84445  \n",
       "4510007  MESH:D000506       Gene     9632  \n",
       "4510008  MESH:D000506       Gene     8774  \n",
       "4510009  MESH:D000506       Gene   131578  \n",
       "...               ...        ...      ...  \n",
       "4545465  MESH:D005923       Gene       81  \n",
       "4545466  MESH:D005923       Gene     7436  \n",
       "4545467  MESH:D005923       Gene     1286  \n",
       "4545468  MESH:D005923       Gene     5054  \n",
       "4545469  MESH:D005923       Gene     4868  \n",
       "\n",
       "[90128 rows x 13 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diseases = pd.concat([GNBR_diseases,Hetionet_diseases,DrugBank_diseases])\n",
    "diseases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "mesh = pd.read_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/disease_terms.csv\")\n",
    "mesh[\"SubjectID\"]= mesh[\"SubjectID\"].str.upper()\n",
    "mesh_diseases = pd.merge(mesh,diseases,on = \"SubjectID\")\n",
    "mesh_diseases = mesh_diseases[[\"SubjectID\",\"Term\"]]\n",
    "mesh_diseases = mesh_diseases.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "disease_mesh_dict = {k : g[\"Term\"].tolist() for k,g in mesh_diseases.groupby(\"SubjectID\")}\n",
    "sim_scores = []\n",
    "\n",
    "for combination in itertools.combinations(mesh_diseases[\"SubjectID\"].unique(),2):\n",
    "    disease1 = combination[0]\n",
    "    disease2 = combination[1]\n",
    "\n",
    "    term1 = disease_mesh_dict[disease1]\n",
    "    term2 = disease_mesh_dict[disease2]\n",
    "\n",
    "    c = set(term1).intersection(term2)\n",
    "    u = set(term2).union(term2)\n",
    "\n",
    "    sim_score = len(c) / float(len(u))\n",
    "    sim_scores.append([disease1,disease2,sim_score])\n",
    "disease_pheno_df = pd.DataFrame(sim_scores, columns = [\"Disease1\", \"Disease2\", \"PHENO-SIM\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "disease_pheno_df.to_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/output/disease-pheno.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_to_pheno = pd.read_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/genes_to_phenotype.tsv\",on_bad_lines=\"skip\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>disease-ID for link</th>\n",
       "      <th>HPO-Term-ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hypoplasia of the uterus</td>\n",
       "      <td>OMIM:614129</td>\n",
       "      <td>HP:0000013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hypergonadotropic hypogonadism</td>\n",
       "      <td>OMIM:614129</td>\n",
       "      <td>HP:0000815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>primary amenorrhea</td>\n",
       "      <td>OMIM:614129</td>\n",
       "      <td>HP:0000786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>autosomal recessive inheritance</td>\n",
       "      <td>OMIM:614129</td>\n",
       "      <td>HP:0000007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>microcephaly</td>\n",
       "      <td>OMIM:614129</td>\n",
       "      <td>HP:0000252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268978</th>\n",
       "      <td>functional motor deficit</td>\n",
       "      <td>ORPHA:98934</td>\n",
       "      <td>HP:0004302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268979</th>\n",
       "      <td>personality changes</td>\n",
       "      <td>ORPHA:98934</td>\n",
       "      <td>HP:0000751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268980</th>\n",
       "      <td>memory impairment</td>\n",
       "      <td>ORPHA:98934</td>\n",
       "      <td>HP:0002354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268981</th>\n",
       "      <td>dementia</td>\n",
       "      <td>ORPHA:98934</td>\n",
       "      <td>HP:0000726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268982</th>\n",
       "      <td>cerebral cortical atrophy</td>\n",
       "      <td>ORPHA:98934</td>\n",
       "      <td>HP:0002120</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>268983 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   name disease-ID for link HPO-Term-ID\n",
       "0              hypoplasia of the uterus         OMIM:614129  HP:0000013\n",
       "1        hypergonadotropic hypogonadism         OMIM:614129  HP:0000815\n",
       "2                    primary amenorrhea         OMIM:614129  HP:0000786\n",
       "3       autosomal recessive inheritance         OMIM:614129  HP:0000007\n",
       "4                          microcephaly         OMIM:614129  HP:0000252\n",
       "...                                 ...                 ...         ...\n",
       "268978         functional motor deficit         ORPHA:98934  HP:0004302\n",
       "268979              personality changes         ORPHA:98934  HP:0000751\n",
       "268980                memory impairment         ORPHA:98934  HP:0002354\n",
       "268981                         dementia         ORPHA:98934  HP:0000726\n",
       "268982        cerebral cortical atrophy         ORPHA:98934  HP:0002120\n",
       "\n",
       "[268983 rows x 3 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_to_pheno[\"HPO-Term-Name\"] = genes_to_pheno[\"HPO-Term-Name\"].str.lower()\n",
    "genes_to_pheno = genes_to_pheno.rename(columns = {\"HPO-Term-Name\" : \"name\"})\n",
    "genes_to_pheno = genes_to_pheno[[\"name\", \"disease-ID for link\", \"HPO-Term-ID\"]]\n",
    "genes_to_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>d</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>mesh:D053559</td>\n",
       "      <td>corneal dystrophy, juvenile epithelial of mees...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>mesh:D015715</td>\n",
       "      <td>corneal edema</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>mesh:D055954</td>\n",
       "      <td>corneal endothelial cell loss</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>mesh:D065306</td>\n",
       "      <td>corneal injuries</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mesh:D016510</td>\n",
       "      <td>corneal neovascularization</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4995</th>\n",
       "      <td>mesh:D008363</td>\n",
       "      <td>alpha-mannosidosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4996</th>\n",
       "      <td>mesh:D017085</td>\n",
       "      <td>alpha-thalassemia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4997</th>\n",
       "      <td>mesh:D044905</td>\n",
       "      <td>beta-mannosidosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4998</th>\n",
       "      <td>mesh:D017086</td>\n",
       "      <td>beta-thalassemia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4999</th>\n",
       "      <td>mesh:D055538</td>\n",
       "      <td>delta-thalassemia</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5000 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 d                                               name\n",
       "0     mesh:D053559  corneal dystrophy, juvenile epithelial of mees...\n",
       "1     mesh:D015715                                      corneal edema\n",
       "2     mesh:D055954                      corneal endothelial cell loss\n",
       "3     mesh:D065306                                   corneal injuries\n",
       "4     mesh:D016510                         corneal neovascularization\n",
       "...            ...                                                ...\n",
       "4995  mesh:D008363                                 alpha-mannosidosis\n",
       "4996  mesh:D017085                                  alpha-thalassemia\n",
       "4997  mesh:D044905                                  beta-mannosidosis\n",
       "4998  mesh:D017086                                   beta-thalassemia\n",
       "4999  mesh:D055538                                  delta-thalassemia\n",
       "\n",
       "[5000 rows x 2 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mesh_diseases_vocab = pd.read_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/MESH_diseases_vocab.tsv\", sep = \"\\t\")\n",
    "mesh_diseases_vocab[\"name\"] = mesh_diseases_vocab[\"name\"].str.lower()\n",
    "mesh_diseases_vocab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SubjectID</th>\n",
       "      <th>name</th>\n",
       "      <th>disease-ID for link</th>\n",
       "      <th>HPO-Term-ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MESH:D016510</td>\n",
       "      <td>corneal neovascularization</td>\n",
       "      <td>ORPHA:567</td>\n",
       "      <td>HP:0011496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MESH:D016510</td>\n",
       "      <td>corneal neovascularization</td>\n",
       "      <td>OMIM:175780</td>\n",
       "      <td>HP:0011496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MESH:D016510</td>\n",
       "      <td>corneal neovascularization</td>\n",
       "      <td>ORPHA:567</td>\n",
       "      <td>HP:0011496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MESH:D016510</td>\n",
       "      <td>corneal neovascularization</td>\n",
       "      <td>ORPHA:567</td>\n",
       "      <td>HP:0011496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MESH:D016510</td>\n",
       "      <td>corneal neovascularization</td>\n",
       "      <td>OMIM:617315</td>\n",
       "      <td>HP:0011496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59075</th>\n",
       "      <td>MESH:D015043</td>\n",
       "      <td>zollinger-ellison syndrome</td>\n",
       "      <td>ORPHA:97279</td>\n",
       "      <td>HP:0002044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59076</th>\n",
       "      <td>MESH:D015043</td>\n",
       "      <td>zollinger-ellison syndrome</td>\n",
       "      <td>ORPHA:652</td>\n",
       "      <td>HP:0002044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59077</th>\n",
       "      <td>MESH:D015043</td>\n",
       "      <td>zollinger-ellison syndrome</td>\n",
       "      <td>OMIM:131100</td>\n",
       "      <td>HP:0002044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59078</th>\n",
       "      <td>MESH:D015043</td>\n",
       "      <td>zollinger-ellison syndrome</td>\n",
       "      <td>OMIM:248700</td>\n",
       "      <td>HP:0002044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59079</th>\n",
       "      <td>MESH:D015043</td>\n",
       "      <td>zollinger-ellison syndrome</td>\n",
       "      <td>ORPHA:97279</td>\n",
       "      <td>HP:0002044</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59080 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          SubjectID                        name disease-ID for link  \\\n",
       "0      MESH:D016510  corneal neovascularization           ORPHA:567   \n",
       "1      MESH:D016510  corneal neovascularization         OMIM:175780   \n",
       "2      MESH:D016510  corneal neovascularization           ORPHA:567   \n",
       "3      MESH:D016510  corneal neovascularization           ORPHA:567   \n",
       "4      MESH:D016510  corneal neovascularization         OMIM:617315   \n",
       "...             ...                         ...                 ...   \n",
       "59075  MESH:D015043  zollinger-ellison syndrome         ORPHA:97279   \n",
       "59076  MESH:D015043  zollinger-ellison syndrome           ORPHA:652   \n",
       "59077  MESH:D015043  zollinger-ellison syndrome         OMIM:131100   \n",
       "59078  MESH:D015043  zollinger-ellison syndrome         OMIM:248700   \n",
       "59079  MESH:D015043  zollinger-ellison syndrome         ORPHA:97279   \n",
       "\n",
       "      HPO-Term-ID  \n",
       "0      HP:0011496  \n",
       "1      HP:0011496  \n",
       "2      HP:0011496  \n",
       "3      HP:0011496  \n",
       "4      HP:0011496  \n",
       "...           ...  \n",
       "59075  HP:0002044  \n",
       "59076  HP:0002044  \n",
       "59077  HP:0002044  \n",
       "59078  HP:0002044  \n",
       "59079  HP:0002044  \n",
       "\n",
       "[59080 rows x 4 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mappings = pd.merge(mesh_diseases_vocab,genes_to_pheno,on = \"name\")\n",
    "mappings[\"d\"] = mappings[\"d\"].str.upper()\n",
    "mappings = mappings.rename(columns={\"d\" : \"SubjectID\"})\n",
    "mappings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SubjectID</th>\n",
       "      <th>HPO-Term-ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MESH:D005901</td>\n",
       "      <td>HP:0000501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MESH:D005901</td>\n",
       "      <td>HP:0000501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MESH:D005901</td>\n",
       "      <td>HP:0000501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MESH:D005901</td>\n",
       "      <td>HP:0000501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MESH:D005901</td>\n",
       "      <td>HP:0000501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3284778</th>\n",
       "      <td>MESH:D014820</td>\n",
       "      <td>HP:0001045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3284779</th>\n",
       "      <td>MESH:D014820</td>\n",
       "      <td>HP:0001045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3284780</th>\n",
       "      <td>MESH:D014820</td>\n",
       "      <td>HP:0001045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3284781</th>\n",
       "      <td>MESH:D014820</td>\n",
       "      <td>HP:0001045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3284782</th>\n",
       "      <td>MESH:D014820</td>\n",
       "      <td>HP:0001045</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3284783 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            SubjectID HPO-Term-ID\n",
       "0        MESH:D005901  HP:0000501\n",
       "1        MESH:D005901  HP:0000501\n",
       "2        MESH:D005901  HP:0000501\n",
       "3        MESH:D005901  HP:0000501\n",
       "4        MESH:D005901  HP:0000501\n",
       "...               ...         ...\n",
       "3284778  MESH:D014820  HP:0001045\n",
       "3284779  MESH:D014820  HP:0001045\n",
       "3284780  MESH:D014820  HP:0001045\n",
       "3284781  MESH:D014820  HP:0001045\n",
       "3284782  MESH:D014820  HP:0001045\n",
       "\n",
       "[3284783 rows x 2 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "disease_to_hpo = pd.merge(mappings,diseases, on = \"SubjectID\")\n",
    "disease_to_hpo = disease_to_hpo[[\"SubjectID\",\"HPO-Term-ID\"]]\n",
    "disease_to_hpo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "disease_to_hpo[\"SubjectID\"]=disease_to_hpo[\"SubjectID\"].map(lambda d: d)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "disease_to_hpo = disease_to_hpo.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "disease_to_hpo.to_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/disease-hpo.query\",sep = \"\\t\", header = False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "disease_to_hpo.to_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/disease-hpo.tsv\",sep = \"\\t\", header = False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "! if [ ! -f '/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/hp.owl' ]; then gunzip '/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/hp.owl'; fi "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------------------------------------\n",
      "\tSemantic Measures Library Toolkit 0.9.4c\n",
      "----------------------------------------------------------------------\n",
      "Toolkit dedicated to Semantic Measures computation and studies.\n",
      "Website: http://www.semantic-measures-library.org\n",
      "Developer: Sébastien Harispe\n",
      "\n",
      "Please cite: \n",
      "The Semantic Measures Library and Toolkit: \n",
      "fast computation of semantic similarity and relatedness \n",
      "using biomedical ontologies.\n",
      "Sébastien Harispe*, Sylvie Ranwez, Stefan Janaqi, Jacky Montmain\n",
      "Bioinformatics 2013; doi: 10.1093/bioinformatics/btt581\n",
      "----------------------------------------------------------------------\n",
      "Args [-t, sm]\n",
      "Args Tool    [-xmlconf, /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/hpo.conf]\n",
      "Loading SM Tool\n",
      "----------------------------------------------------------------------\n",
      "\tSemantic Measures Computer 0.9.4c\n",
      "----------------------------------------------------------------------\n",
      "Tool used to compute Semantic similarity/relatedness considering an\n",
      "XML configuration file or a specific profile dedicated to an ontology.\n",
      "\n",
      "---------------------------------------------------------------\n",
      " Processing XML configuration /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/hpo.conf\n",
      "---------------------------------------------------------------\n",
      "Loading XML conf from : /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/hpo.conf\n",
      "Setting maximal number of threads to 4\n",
      "add namespace prefix : mesh ref : http://bioonto.de/mesh.owl#\n",
      "add namespace prefix : HP ref : http://purl.obolibrary.org/obo/HP_\n",
      "add namespace prefix : DRUGBANK ref : http://bio2rdf.org/drugbank:\n",
      "generic configuration loaded \n",
      "Loading sm Configuration : /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/hpo.conf\n",
      "1 IC configurations loaded \n",
      "1 pairwise measure configurations loaded \n",
      "1 groupwise measure configurations  loaded \n",
      "Configuration loaded... \n",
      "Loading Graph http://bio2rdf.org/mesh:\n",
      "-------------------------------------\n",
      " Loading DATA\n",
      "-------------------------------------\n",
      "-------------------------------------\n",
      " RDF Loader\n",
      "-------------------------------------\n",
      "Populate graph http://bio2rdf.org/mesh: from /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/hp.owl\n",
      "Parser loaded for: RDF/XML (mimeTypes=application/rdf+xml, application/xml; ext=rdf, rdfs, owl, xml)\n",
      "Parsing RDF file...\n",
      "Start Process\n",
      "Ending Process 95027 statements loaded \n",
      "vertices: 38240\n",
      "edges   : 95027\n",
      "Skipped (statement involving non URI ressources) : 989777\n",
      "Graph http://bio2rdf.org/mesh: populated by RDF data \n",
      "-------------------------------------\n",
      "-------------------------------------\n",
      "Loading Annotations using TSV loader.\n",
      "-------------------------------------\n",
      "file            /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/disease-hpo.tsv\n",
      "Skipping header false\n",
      "predicate       'http://www.w3.org/1999/02/22-rdf-syntax-ns#type'\n",
      "Number of lines skipped   0\n",
      "Number of lines processed 26\n",
      "Number of statements loaded 26\n",
      "TSV specification loaded.\n",
      "-------------------------------------\n",
      "-------------------------------------\n",
      " Applying actions\n",
      "-------------------------------------\n",
      "-------------------------------------\n",
      "Rerooting\n",
      "-------------------------------------\n",
      "Fetching root node, uri: __FICTIVE__\n",
      "No root node explicitly specified using 'root_uri' parameter. Set root : http://www.w3.org/2002/07/owl#Thing\n",
      "Rooting taxonomic Graph using http://www.w3.org/2002/07/owl#Thing\n",
      "Number of roots 400\n",
      "Starting process from 19275 vertices\n",
      "isDag : false\n",
      "current path :Path: \n",
      "\thttp://purl.obolibrary.org/obo/PR_P17679 -- http://www.w3.org/2000/01/rdf-schema#subClassOf -- http://purl.obolibrary.org/obo/PR_000029032\n",
      "\thttp://purl.obolibrary.org/obo/PR_000029032 -- http://www.w3.org/2000/01/rdf-schema#subClassOf -- http://purl.obolibrary.org/obo/PR_000036194\n",
      "\thttp://purl.obolibrary.org/obo/PR_000036194 -- http://www.w3.org/2000/01/rdf-schema#subClassOf -- http://purl.obolibrary.org/obo/PR_000000001\n",
      "\thttp://purl.obolibrary.org/obo/PR_000000001 -- http://www.w3.org/2000/01/rdf-schema#subClassOf -- http://purl.obolibrary.org/obo/CHEBI_36080\n",
      "\thttp://purl.obolibrary.org/obo/CHEBI_36080 -- http://www.w3.org/2000/01/rdf-schema#subClassOf -- http://purl.obolibrary.org/obo/PR_000000001\n",
      "Cycle detected adding : http://purl.obolibrary.org/obo/CHEBI_36080 -- http://www.w3.org/2000/01/rdf-schema#subClassOf -- http://purl.obolibrary.org/obo/PR_000000001 to path\n",
      "\n",
      "\n",
      "[Error] Error during rerooting: Underlying graph build from  contraint Walconstraint\n",
      "\tAcceptedWalkIN: \n",
      "\tAcceptedWalkOUT:\n",
      "\t\thttp://www.w3.org/2000/01/rdf-schema#subClassOf\n",
      "is not a DAG\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!java -jar /Users/elifozkan/Desktop/sml-toolkit-0.9.4c.jar -t sm -xmlconf /Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/hpo.conf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Subject</th>\n",
       "      <th>Predicate</th>\n",
       "      <th>Object</th>\n",
       "      <th>Source1</th>\n",
       "      <th>Source2</th>\n",
       "      <th>Source3</th>\n",
       "      <th>Source4</th>\n",
       "      <th>Source5</th>\n",
       "      <th>Source6</th>\n",
       "      <th>SubjectType</th>\n",
       "      <th>SubjectID</th>\n",
       "      <th>ObjectType</th>\n",
       "      <th>ObjectID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4510005</th>\n",
       "      <td>Disease::MESH:D000506</td>\n",
       "      <td>Hetionet::DdG::Disease:Gene</td>\n",
       "      <td>Gene::26059</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D000506</td>\n",
       "      <td>Gene</td>\n",
       "      <td>26059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4510006</th>\n",
       "      <td>Disease::MESH:D000506</td>\n",
       "      <td>Hetionet::DdG::Disease:Gene</td>\n",
       "      <td>Gene::84445</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D000506</td>\n",
       "      <td>Gene</td>\n",
       "      <td>84445</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4510007</th>\n",
       "      <td>Disease::MESH:D000506</td>\n",
       "      <td>Hetionet::DdG::Disease:Gene</td>\n",
       "      <td>Gene::9632</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D000506</td>\n",
       "      <td>Gene</td>\n",
       "      <td>9632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4510008</th>\n",
       "      <td>Disease::MESH:D000506</td>\n",
       "      <td>Hetionet::DdG::Disease:Gene</td>\n",
       "      <td>Gene::8774</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D000506</td>\n",
       "      <td>Gene</td>\n",
       "      <td>8774</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4510009</th>\n",
       "      <td>Disease::MESH:D000506</td>\n",
       "      <td>Hetionet::DdG::Disease:Gene</td>\n",
       "      <td>Gene::131578</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D000506</td>\n",
       "      <td>Gene</td>\n",
       "      <td>131578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4545465</th>\n",
       "      <td>Disease::MESH:D005923</td>\n",
       "      <td>Hetionet::DaG::Disease:Gene</td>\n",
       "      <td>Gene::81</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D005923</td>\n",
       "      <td>Gene</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4545466</th>\n",
       "      <td>Disease::MESH:D005923</td>\n",
       "      <td>Hetionet::DaG::Disease:Gene</td>\n",
       "      <td>Gene::7436</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D005923</td>\n",
       "      <td>Gene</td>\n",
       "      <td>7436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4545467</th>\n",
       "      <td>Disease::MESH:D005923</td>\n",
       "      <td>Hetionet::DaG::Disease:Gene</td>\n",
       "      <td>Gene::1286</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D005923</td>\n",
       "      <td>Gene</td>\n",
       "      <td>1286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4545468</th>\n",
       "      <td>Disease::MESH:D005923</td>\n",
       "      <td>Hetionet::DaG::Disease:Gene</td>\n",
       "      <td>Gene::5054</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D005923</td>\n",
       "      <td>Gene</td>\n",
       "      <td>5054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4545469</th>\n",
       "      <td>Disease::MESH:D005923</td>\n",
       "      <td>Hetionet::DaG::Disease:Gene</td>\n",
       "      <td>Gene::4868</td>\n",
       "      <td>[Drugbank]</td>\n",
       "      <td>[Hetionet]</td>\n",
       "      <td>[GNBR]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Disease</td>\n",
       "      <td>MESH:D005923</td>\n",
       "      <td>Gene</td>\n",
       "      <td>4868</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>90128 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Subject                    Predicate        Object  \\\n",
       "4510005  Disease::MESH:D000506  Hetionet::DdG::Disease:Gene   Gene::26059   \n",
       "4510006  Disease::MESH:D000506  Hetionet::DdG::Disease:Gene   Gene::84445   \n",
       "4510007  Disease::MESH:D000506  Hetionet::DdG::Disease:Gene    Gene::9632   \n",
       "4510008  Disease::MESH:D000506  Hetionet::DdG::Disease:Gene    Gene::8774   \n",
       "4510009  Disease::MESH:D000506  Hetionet::DdG::Disease:Gene  Gene::131578   \n",
       "...                        ...                          ...           ...   \n",
       "4545465  Disease::MESH:D005923  Hetionet::DaG::Disease:Gene      Gene::81   \n",
       "4545466  Disease::MESH:D005923  Hetionet::DaG::Disease:Gene    Gene::7436   \n",
       "4545467  Disease::MESH:D005923  Hetionet::DaG::Disease:Gene    Gene::1286   \n",
       "4545468  Disease::MESH:D005923  Hetionet::DaG::Disease:Gene    Gene::5054   \n",
       "4545469  Disease::MESH:D005923  Hetionet::DaG::Disease:Gene    Gene::4868   \n",
       "\n",
       "            Source1     Source2 Source3 Source4 Source5 Source6 SubjectType  \\\n",
       "4510005  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4510006  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4510007  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4510008  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4510009  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "...             ...         ...     ...     ...     ...     ...         ...   \n",
       "4545465  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4545466  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4545467  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4545468  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "4545469  [Drugbank]  [Hetionet]  [GNBR]     NaN     NaN     NaN     Disease   \n",
       "\n",
       "            SubjectID ObjectType ObjectID  \n",
       "4510005  MESH:D000506       Gene    26059  \n",
       "4510006  MESH:D000506       Gene    84445  \n",
       "4510007  MESH:D000506       Gene     9632  \n",
       "4510008  MESH:D000506       Gene     8774  \n",
       "4510009  MESH:D000506       Gene   131578  \n",
       "...               ...        ...      ...  \n",
       "4545465  MESH:D005923       Gene       81  \n",
       "4545466  MESH:D005923       Gene     7436  \n",
       "4545467  MESH:D005923       Gene     1286  \n",
       "4545468  MESH:D005923       Gene     5054  \n",
       "4545469  MESH:D005923       Gene     4868  \n",
       "\n",
       "[90128 rows x 13 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diseases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "disease_to_hpo.to_csv(\"/Users/elifozkan/Desktop/translator-openpredict/data/drkg/feature_related/output/disease-hpo.csv\", index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "6ff0ae482b5d3ed02d07ac49d1cc8913c1946d500f10ba0f102f82171bb9657d"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
